timon/run.py

```diff
@@ -24,16 +24,28 @@
 logger = logging.getLogger(__name__)
 
 
-async def ask_exit(signame, cfg):
+async def ask_exit(signame, stopevent, cfg):
     """
     This code is an attempt to help performing a
-    clean shutdown
-    At moment, only stop plugins and stop the dbstore
+    clean shutdown.
+    First time the signal is catched, set an asyncio event to
+    True that permits to not relaunch the loop and cleanly stop after
+    probes process.
+    Second time the signal is catched, abruptly stop timon (
+    but cleanly stop plugins and the dbstore to avoid loosing data)
     """
-    logger.info("got signal %r: will exit", signame)
-    await cfg.stop_plugins()
-    cfg.stop_dbstore()
-    sys.exit()
+    logger.info("got signal %r: will ask exit", signame)
+    if not stopevent.is_set():
+        stopevent.set()
+    else:
+        logger.warning("Second stop signal: Force exit")
+        try:
```

Comment:
is there any issue if the first ctrl-C stopped already the plugins and we call stop_plugins with the second ctrl-C ?
I think it should be OK. but want to be sure

Reply:
Mmm, at moment no, there's no problem with the only plugin we have, and in the future if there's an error when we call stop_pluins twice, it's to this function to be more robust not to the shutdown func (with for exemple no call to the plugin's stop func if the plugin is stopped)